assets/terraform-modules/packet/flatcar-linux/kubernetes/workers/cl/worker.yaml.tmpl

```diff
@@ -82,12 +82,9 @@ systemd:
         ExecStartPre=/usr/bin/bash -c "grep 'certificate-authority-data' /etc/kubernetes/kubeconfig | awk '{print $2}' | base64 -d > /etc/kubernetes/ca.crt"
         ExecStartPre=-/usr/bin/rkt rm --uuid-file=/var/cache/kubelet-pod.uuid
         ExecStartPre=/etc/kubernetes/configure-kubelet-cgroup-driver
-        # TODO: Workaround until https://github.com/coreos/afterburn/pull/358
-        # makes it into Flatcar. Then we can read COREOS_PACKET_IPV4_PRIVATE_GATEWAY_0
-        # from /run/metadata/flatcar and disable the template conditionals below.
         ExecStart=/bin/sh -c \
             "%{~ if bgp_node_labels != "" ~}
-            BGP_PEER_ADDRESS=$(ip route | grep '10.0.0.0/8' | awk {'print $3'}); \
+            BGP_PEER_ADDRESS=$(curl -s https://metadata.packet.net/metadata | jq -r .bgp_neighbors[0].peer_ips[0]); \
```

Comment:
What if curl time out or hangs for a very long time? I think the content of the variable might not be a valid peer and we probably should fail?

I think adding these parameters might help (is not nice, though):  `curl --retry-delay 1 --retry 60 --retry-connrefused --retry-max-time 60 --connect-timeout 20`

(Taken from https://github.com/kinvolk/vpn/blob/923a75d6713690e811fb003a1bf01a9e3398f307/bird/startup.sh#L9, this repo is not public for other seeing it. Sorry :()

That will only help to not hang, but we still need to handle the failure scenario.

Reply:
Good point. We're facing another problem though: sometimes `curl` runs before the BGP session got created by Terraform, in which case the value becomes `null`. I'm trying to avoid having to wrap this in a retry script.

Reply:
I've added a "wait for" script which handles retries. I've added `--connect-timeout` though.

Reply:
Actually we may want some flags in the actual command which gets the data. Fixing.